### PR TITLE
Da#312,#241 cli dependency downloader and CB Export

### PR DESCRIPTION
**Pull request by lokesh-couchbase**
This PR focusses on adding the CLI dependency downloader as well as support CB Export

### CODE REVIEW

File: scripts/dependency_downloader.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    const cbImportDir = path.join(toolsPath, cbImport.getInstallationPath());
+    if (tools[Type.CB_IMPORT].status === ToolStatus.NOT_AVAILABLE
+        && !isInstalled(toolsPath, downloads.get(TOOL_IMPORT_EXPORT), Type.CB_EXPORT)) {
+        // avoiding 2 threads to install the same thing at the same time
+        console.log("Downloading CB Import/Export. The feature will be automatically enabled when the download is complete.");
+        tools[Type.CB_EXPORT].status = (ToolStatus.DOWNLOADING);
+        tools[Type.CB_IMPORT].status = (ToolStatus.DOWNLOADING);
+        downloadAndUnzip(cbImportDir, cbImport);
+    } else {
+        Log.debug("CB Import/Export is already installed");
```
Comment (AayushTyagi1) [inline scripts/dependency_downloader.js:198]: Why we are not using logger

File: src/extension.ts [hunk 2]
```diff
@@ -50,6 +57,9 @@ import { handleActiveEditorChange } from "./handlers/handleActiveTextEditorChang
 import { QueryWorkbench } from "./workbench/queryWorkbench";
 import { WorkbenchWebviewProvider } from "./workbench/workbenchWebviewProvider";
 import { fetchClusterOverview } from "./pages/overviewCluster/overviewCluster";
+// import { Worker, isMainThread, parentPort } from 'worker_threads';
```
Comment (AayushTyagi1) [inline src/extension.ts:60]: Why commented imports

File: src/util/DependencyDownloaderUtils/fileUtils.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as childProcess from 'child_process';
+import OSUtil from './OSUtils';
+import { logger } from '../../logger/logger';
+
```
Comment (AayushTyagi1) [inline src/util/DependencyDownloaderUtils/fileUtils.ts:6]: Change DependencyDownloaderUtils -> dependencyDownloaderUtils
Change OverviewClusterUtils -> overviewClusterUtils

File: src/webViews/tools/dataExport.webview.ts [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        
+                    // Clear existing options in the scope dropdown
+                    scopeDropdown.innerHTML = '';
+                    const selectAllOption = document.createElement('option');
+                    selectAllOption.value = 'All Scopes';
+                    selectAllOption.text = 'Select All';
+                    scopeDropdown.appendChild(selectAllOption);
+                    
+                    // Add scope options
+                    scopesData.forEach((scope) => {
```
Comment (AayushTyagi1) [inline src/webViews/tools/dataExport.webview.ts:276]: Change the icon from couchbase icon to any suitable icon

Comment (AayushTyagi1) [reply src/webViews/tools/dataExport.webview.ts:276]: Could you please change button color to Couchbase Red color

File: src/tools/CBExport.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            if (verbose) {
+                cmd.push("-v");
+            }
+
+            const terminal = vscode.window.createTerminal("CBExport",);
+            let text = cmd.join(" ");
+            terminal.sendText(text);
+            terminal.show();
+
+            // const process = child_process.spawnSync(cmd[0], cmd.slice(1), {
```
Comment (AayushTyagi1) [inline src/tools/CBExport.ts:115]: Can you please remove the commented code?

File: src/tools/CBExport.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            //     );
+            // } else {
+            //     console.log(`File ${fileName} was exported successfully`);
+            //     console.log(`File ${fullPath} saved successfully.`);
+            // }
+        } catch (error) {
+            console.error("An error occurred while trying to export the dataset");
+            console.error(error);
+        }
+        // });
```
Comment (AayushTyagi1) [inline src/tools/CBExport.ts:132]: Here too

File: src/extension.ts [hunk 7]
```diff
@@ -443,13 +484,19 @@ export function activate(context: vscode.ExtensionContext) {
 
   subscriptions.push(
     vscode.commands.registerCommand(
-      Commands.queryContext,
-      () => {
-        fetchQueryContext(workbench, context);
+      "vscode-couchbase.tools.dataExport",
```
Comment (AayushTyagi1) [inline src/extension.ts:487]: Can you put the command in Commands

File: src/pages/Tools/DataExport/dataExport.ts [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+import { Memory } from "../../../util/util";
+import { Constants } from "../../../util/constants";
+import { logger } from "../../../logger/logger";
+import * as path from "path";
+import { getLoader } from "../../../webViews/loader.webview";
+import { dataExportWebview } from "../../../webViews/tools/dataExport.webview";
+import { IConnection } from "../../../types/IConnection";
+
+const getScopes = async (bucketId: string, connection: IConnection) => {
+  let scopes = await connection.cluster
```
Comment (AayushTyagi1) [inline src/pages/Tools/DataExport/dataExport.ts:17]: use const

File: src/webViews/tools/dataExport.webview.ts (hunk 4 above)
Comment (lokesh-couchbase) [reply src/webViews/tools/dataExport.webview.ts:325]: done

File: src/util/DependencyDownloaderUtils/fileUtils.ts (hunk 3 above)
Comment (lokesh-couchbase) [reply src/util/DependencyDownloaderUtils/fileUtils.ts:6]: done

File: src/tools/CBExport.ts (hunk 6 above)
Comment (lokesh-couchbase) [reply src/tools/CBExport.ts]: done

File: src/tools/CBExport.ts (hunk 5 above)
Comment (lokesh-couchbase) [reply src/tools/CBExport.ts]: done

**Comment by AayushTyagi1** [comment]
LGTM, 
Please check if the advance setting design needs Accordian.

**Review by AayushTyagi1 — APPROVED** [review]

**Pull request merged**